pkg/compose/create.go

```diff
@@ -874,8 +866,8 @@ func (s *composeService) buildContainerVolumes(
 			v := findVolumeByTarget(service.Volumes, m.Target)
 			vol := findVolumeByName(p.Volumes, m.Source)
 			if v != nil && vol != nil {
-				if _, ok := vol.DriverOpts["device"]; ok && vol.Driver == "local" && vol.DriverOpts["o"] == "bind" {
```

Comment:
removed this earlier fix, as with `volumeRequiresMountAPI` we will prefer `bind` for all volumes without advanced mount options, not just those created with `o=bind`